sys-boot/refind/refind-0.14.0.2-r1.ebuild

```diff
@@ -0,0 +1,183 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit toolchain-funcs secureboot
+
+DESCRIPTION="The UEFI Boot Manager by Rod Smith"
+HOMEPAGE="https://www.rodsbooks.com/refind/"
+SRC_URI="mirror://sourceforge/project/${PN}/${PV}/${PN}-src-${PV}.tar.gz"
+
+LICENSE="BSD GPL-2 GPL-3 FDL-1.3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+FS_USE="btrfs +ext2 +ext4 hfs +iso9660 ntfs reiserfs"
+IUSE="${FS_USE} custom-cflags doc"
+
+DEPEND="sys-boot/gnu-efi"
+
+# for ld.bfd and objcopy
+BDEPEND="sys-devel/binutils"
+
+DOCS=( README.txt NEWS.txt )
+
+PATCHES=( "${FILESDIR}"/${P}-clang.patch )
+
+checktools() {
+	if [[ ${MERGE_TYPE} != "binary" ]]; then
+		# 832018
+		export LD="$(tc-getLD)"
+		tc-ld-force-bfd
+		# tc-ld-force-bfd does not work in all cases, so try to also fix LD
+		LD="${LD/.lld/.bfd}"
+		tc-ld-is-lld "${LD}" && die "Linking with lld produces broken executables and may lead to unbootable system"
+
+		# 732256
+		# llvm-objcopy does not support EFI target, try to use binutils objcopy or fail
+		export OBJCOPY="$(tc-getOBJCOPY)"
+		OBJCOPY="${OBJCOPY/llvm-/}"
+		LANG=C LC_ALL=C "${OBJCOPY}" --help | grep -q '\<pei-' || die "${OBJCOPY} (objcopy) does not support EFI target"
+	fi
+}
+
+pkg_pretend() {
+	if use custom-cflags; then
+		ewarn
+		ewarn "You have enabled building with USE=custom-cflags. Be aware that"
+		ewarn "using this can result in EFI binaries that fail to run and may"
+		ewarn "fail to build at all. This is strongly advised against by upstream."
+		ewarn
+		ewarn "See https://bugs.gentoo.org/598587#c3 for more information"
+		ewarn
+	fi
+
+	checktools
+}
+
+pkg_setup() {
+	if use x86; then
+		export EFIARCH=ia32
+		export BUILDARCH=ia32
+	elif use amd64; then
+		export EFIARCH=x64
+		export BUILDARCH=x86_64
+	fi
+	secureboot_pkg_setup
+
+	# this does not only check, but also exports LD and OBJCOPY
+	checktools
+}
+
+src_prepare() {
+	default
+
+	# bug 598647 - PIE not supported
+	sed -e '/^CFLAGS/s/$/ -fno-PIE/' -i Make.common || die
+	sed -e '1 i\.NOTPARALLEL:' -i filesystems/Makefile || die
+
+	# 881131, 832018
+	if tc-is-clang; then
+		sed -e 's/-fno-tree-loop-distribute-patterns//' -i Make.common || die
```

Comment:
I suspect this is a bit dubious to do at all (i.e. we should sed it unconditionally).

See my remarks in https://forums.gentoo.org/viewtopic-t-1163129-start-0-postdays-0-postorder-asc-highlight-refind.html (the whole thread is also interesting).

Reply:
I haven't seen the forum thread, I'll read it.  I'll try to sed it unconditionally and test if refind works ok after that.  I assumed it's important for gcc, but it's better to test.

Reply:
gcc 12 without `-fno-tree-loop-distribute-patterns` errors out, but adding `-ffreestanding` instead works, and looks like a better solution.
clang works too with `-ffreestanding`, except it generates an extra link error because or `_FORTIFY_SOURCE` (which is added by default by `sys-devel/clang-common` on my machine, not sure if it is like this for everyone).  I added an #undef in the patch to fix this.